File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        string ch;
        int a,b;

        for (int i = 0; i < tokens.size(); i++) {
            ch = tokens[i];

            if (ch == "+") {
               b =  st.top();
               st.pop();
               a = st.top();
               st.pop();
                st.push(a + b);
            } else if (ch == "-") {
                 b =  st.top();
               st.pop();
               a = st.top();
               st.pop();
                st.push(a - b);
            } else if (ch == "*") {
                 b =  st.top();
               st.pop();
               a = st.top();
               st.pop();
                st.push(a * b);
            } else if (ch == "/") {
                 b =  st.top();
               st.pop();
               a = st.top();
               st.pop();
                st.push(a / b);
            } else {
                st.push(stoi(ch));
            }
        }
        int val = st.top();
        return val;
    }
};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def calculate(self,num1,num2,operand):
        if operand == '+':
            return num1 + num2
        elif operand == '-':
            return num1-num2
        elif operand == '*':
            return num1*num2
        else:
            return int(num1 / num2)
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        
        for token in tokens:
            if token in "+-*/":
                num1 = stack.pop()
                num2 = stack.pop()
                stack.append(self.calculate(num2,num1,token))
            else:
                stack.append(int(token))
        return stack.pop()                